lib/src/over_react_test/react_util.dart

```diff
@@ -481,8 +481,35 @@ Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey
 ///     </div>
 ///
 ///     queryAllByTestId(renderedInstance, 'value'); // returns both `inner` `<div>`s
-List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey}) {
-  return findDomNode(root).querySelectorAll('[$key~="$value"]');
+List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth}) {
+  return _findAllDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: true, depth: shadowDepth);
+}
+
+String _makeTestIdSelector(String value, {String key = defaultTestIdKey}) => '[$key~="$value"]';
+
+dynamic /* Element | List<Element> */ _findAllDeep(Element root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int depth}) {
+  List<Element> nodes = [];
+  var currentDepth = 0;
+  var lightElement = root.querySelector(itemSelector);
+  if (!findMany && lightElement != null){
+    return lightElement;
+  }
+  void recursiveSeek(dynamic _root) {
+    nodes = [...nodes, ..._root.querySelectorAll(itemSelector)];
+    if (!findMany && nodes.isNotEmpty) {
+      return;
+    }
+    // This is kinda gross but I figured this was only ever used in test situations so its probably alright.
+    if ((depth == null || (depth != null && currentDepth < depth)) && searchInShadowDom == true) {
+      var foundShadows = _root.querySelectorAll('*').where((el) => el.shadowRoot != null).map((el) => el.shadowRoot).toList();
+      if (foundShadows.isNotEmpty) {
```

Comment:
This is a depth-first recursive search, so incrementing currentDepth on each call will result in it being invalid after backtracking.

If we want to keep this functionality, it could be passed in as an argument to recursiveSeek instead to fix that issue, and there should be more extensive test coverage.

Reply:
ahhh i see what you mean, hu rumph